scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,425 @@
+                        if not vnd:
+                            elem_item.sig_model_add(id1)
+                        else:
+                            elem_item.vnd_model_add(id1, id2)
+
+                if sig_count > 0:
+                    models_unpack(sig_ptr, elem_item, False)
+                if vnd_count > 0:
+                    models_unpack(vnd_ptr, elem_item, True)
+
+    def extract_comp_data(self, kconfigs, comp):
+        """
+        Extract composition data from .elf file.
+
+        Parameters:
+            kconfigs (dict): A dictionary representing Kconfig options used for the firmware to compile with
+            comp (dict): A Composition Data dictionary class object to be filled in
+
+        Returns:
+            TBA
+        """
+        for addr in self.addrs:
+            self.__read_comp_data(addr, kconfigs, comp)
+
+
+def parse_kconfig(config_path):
+    """
+    Extracts content of '.config' file into a dictionary
+
+    Parameters:
+        config_path (str): Path to '.config' file used by a firmware
+    Returns:
+        dict: A dictionary where key is Kconfig option name and value is the value of the corresponding Kconfig option.
+    """
+    with open(config_path, 'r') as config:
+        configs = {}
+        for line in config:
+            if not line.startswith("CONFIG_"):
+                continue
+            kconfig, value = line.split("=")
+            configs[kconfig] = value.split("\n")[0]
+        return configs
+
+def version_parse(cfg):
+    if 'CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION' not in cfg.keys():
+        print("The CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION Kconfig option not found in .config file")
+        return None
+
+    try:
+        clean_str = cfg['CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION'].replace("+", ".").replace("\"", "")
+        version_list = [int(s) for s in clean_str.split(".") if s.isdigit()]
+        return {
+            "major": version_list[0],
+            "minor": version_list[1],
+            "revision": version_list[2],
+            "build_number": version_list[3],
+        }
+    except Exception:
+        print("Unable to parse CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION Kconfig option")
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("Missing build folder argument")
+        sys.exit(1)
+
+    bin_path = sys.argv[1]
+    file_name_in_zip = 'mesh_metadata.json'
+    zip_path = os.path.abspath(bin_path + '/dfu_application.zip')
+    file_name = 'dfu_application.zip_mesh_metadata.json'
+    metadata_path = os.path.abspath(bin_path + '/' + file_name)
+    elf_path = os.path.abspath(bin_path + '/zephyr.elf')
+    config_path = os.path.abspath(bin_path + '/.config')
+
+
+    zip = ZipFile(zip_path, "a")
+    if file_name_in_zip in zip.namelist():
+        # Mesh metadata already present in zip file
+        sys.exit(0)
+
+    comp = Comp0()
+    kconfigs = parse_kconfig(config_path)
+    elf_parser = CompElfParser(elf_path)
+
+    comp.features_add(kconfigs)
```

Comment:
I'm thinking this should perhaps be called in the parse-elf-function that parses the Elf. That function necessarily already has the `kconfigs` available, it makes more sense for that parse-function to return a fully specified `Comp0` (or, list of `Comp0`s like mentioned yesterday) I think.